test/user.test.js

```diff
@@ -401,4 +401,169 @@ describe('USER TEST SUITE', () => {
         expect(response.body.error).to.equal('Invalid token');
       });
   });
+
+  describe('User Update Validations', () => {
+    it('should return No token provided, when No token is provided',
+      async () => {
+        const response = await chai.request(server)
+          .patch('/api/v1/user');
+        expect(response.status).to.equal(400);
+        expect(response.body.error).to.equal('No token provided');
+      });
+
+
+    it('should return Invalid token, when token is Invalid',
+      async () => {
+        const response = await chai.request(server)
+          .patch('/api/v1/user')
+          .set('Authorization', 'invalidtoken');
+        expect(response.status).to.equal(401);
+        expect(response.body.error).to.equal('Invalid token');
+      });
+
+
+    it('should return User Updated Successfully,with correct parameters',
+      async () => {
+        const newUser2 = {
+          email: 'talkto@gmail.com',
+          username: 'talktoat',
+          password: 'htryuufhfhdgsgggx'
+        };
+        const response = await chai.request(server)
+          .patch('/api/v1/user')
+          .send(newUser2)
+          .set('Authorization', firstUserToken);
+        expect(response.body.message).to.equal('User Updated Successfully');
+        expect(response.body.username).to.equal('talktoat');
+        expect(response.body.email).to.equal('talkto@gmail.com');
+        expect(response.body.isMentor).to.equal(false);
+      });
+
+
+    it('should return username already exist,for existing Username',
+      async () => {
+        const newUser2 = {
+          email: 'talkto2@gmail.com',
+          username: 'talktoat',
+          password: 'tyuiiooooosgffshh'
+        };
+        const response = await chai.request(server)
+          .patch('/api/v1/user')
+          .send(newUser2)
+          .set('Authorization', firstUserToken);
+        expect(response.body.error).to.equal('Username already in use');
+      });
+
+
+    it('should return username must be at least 6 ,when charcters less than 6',
```

Comment:
`Should not allow update when the username is not the permitted username format`